### ISSUE TITLE
Fails When Client Doesn't Send Status Request

### ISSUE THREAD

**Issue by coolGi69** [Duckulus/mc-honeypot#1]
I've been recently testing making a simple server pinger in rust and found this program.\
When testing it, I found a discrepancy with the Nochian server where Nochian servers allow for a status request to be omitted (see [the relevent wiki section](https://wiki.vg/Server_List_Ping#Status_Request)), while Honepot doesn't like it when this request is not present, causing the following error.\
`2024-10-14T15:35:23.368Z ERROR [mc_honeypot::server] Connection reset by peer (os error 104)`

Honeypot also seems to wait until the client sends the ping request (only waiting for the status request), while Nochian servers do.\
[The Wiki](https://wiki.vg/Server_List_Ping#Status_Response) states that:
> Note that Notchian servers will for unknown reasons wait to receive the following [Ping Request](https://wiki.vg/Protocol#Ping_Request) packet for 30 seconds before timing out and sending Response.

but Honeypot doesn't do this atm.

For viewing the packet I send out to replicate this behavior, view the code at https://gitlab.com/coolGi/mc-server-pinger/-/blob/eb13c7d8ddbedeb5e44e2989e436010fe5d875a6/src/pinger.rs#L170

**Comment by Duckulus** [comment]
First of all, thank you very much for pointing out these issue!

As for the server failing to respond when the status request is omitted, this is now fixed. The honeypot now responds with the status and pong response, regardless of whether the status request has been sent or not.

I don't really understand the second point you mention. Could you elaborate on what the expected behavior is and how the honeypots behavior differs from it? Should there be a timeout of 30 seconds if the ping request gets sent after a delay?

**Comment by coolGi69** [comment]
Thanks for the fix!

The 30-second timeout was mainly to stop people from attempting to figure out whether the server is honeypot or not. Any (sane) server scanner wouldn't be setting their timeout at 30 seconds and probably have a better way (and quicker) of checking, so it honestly doesn't really matter.\
Just a suggestion in case :)

If the client sends the handshake and status request without sending the ping request, then the server will respond back with the status response after 30 seconds, rather than instantly when the ping request is sent.

Feel free to close it if you don't want to add it. And thanks again for the quick fix!

**Comment by Duckulus** [comment]
Yeah I see, I don't really feel like fixing it atm, but I'll leave this open since its a valid issue
